### PR TITLE
feat: Bounties pages

### PR DESCRIPTION
**Pull request by nhsz**
This PR adds the `/bounties` pages

- [x] `/bounties` index page
  - [x] RSA bounties
    - [x] .md data source 
    - [x] /index
    - [x] /assumptions
      - [x] footnotes 
    - [x] /bounties
      - [x] footnotes 
    - [x] /concrete-instance-bounties
  - [x] Legendre PRF bounties
    - [x] .md data source
    - [x] /algorithmic-bounties
      - [x] footnotes broken
    - [x] /concrete-instance-bounties
    - [x] /smart-contract-instructions
  - [x] ZK Hash Function Cryptanalysis Bounties
    - [x] .md data source 
  - [x] MiMC Hash Challenge page
    - [x] .md data source
- [x] `Bounty` component

### CODE REVIEW

**Comment by samajammin** [comment]
@nhsz what are those `/public/instances/*.bin` files?

File: src/pages/bounties/legendre-prf/algorithmic-bounties.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+interface Props {
+  content: string;
+}
+
+const LegendrePrfBounties: NextPage<Props> = ({ content }) => {
+  return (
+    <>
+      <PageMetadata
+        title='Legendre PRF algorithmic Bounties'
```
Comment (samajammin) [inline src/pages/bounties/legendre-prf/algorithmic-bounties.tsx:38]: I think `title` & `description` for each bounty page should come from the frontmatter of each respective markdown file, like it does for blog posts.

File: src/pages/bounties/legendre-prf/concrete-instance-bounties.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/* eslint-disable react/no-children-prop */
```
Comment (samajammin) [inline src/pages/bounties/legendre-prf/concrete-instance-bounties.tsx:1]: Do we need a standalone React component for each page here? I was hoping for something similar to the `[slug].tsx` setup you implemented for blog posts, where we just have one React component file serve as a template for all bounty pages. Seems redundant/unnecessary to have a React component for each of these.

**Review by samajammin — CHANGES_REQUESTED** [review]
Deploy looks great! A couple questions/requests on the implementation front. Curious to discuss what's possible here.

File: src/pages/bounties/legendre-prf/algorithmic-bounties.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        description='Bounties on breaking the Legendre PRF.'
+      />
+
+      <main>
+        <Heading as='h1' mb={20}>
+          Legendre PRF algorithmic bounties
```
Comment (samajammin) [inline src/pages/bounties/legendre-prf/algorithmic-bounties.tsx:44]: Same here - I think pretty much all content/copy should be sourced from the markdown file (to make it easier for someone to create a bounty page).

File: src/constants.ts [hunk 4]
```diff
@@ -24,3 +22,20 @@ export const HEAD_TITLE_SHORT = 'EF Cryptography Research';
 
 // posts
 export const POSTS_DIR = 'src/posts';
+
+// bounties data sources
+export const BOUNTIES_DATA_SOURCE = 'src/bounties-data-source';
```
Comment (samajammin) [inline src/constants.ts:27]: Same deal - can we set this programmatically (based on the directory structure & markdown filenames)?

**Comment by nhsz** [comment]
> @nhsz what are those `/public/instances/*.bin` files?

these files https://github.com/dankrad/legendreprf/tree/master/instances

File: src/constants.ts [hunk 5]
```diff
@@ -24,3 +22,20 @@ export const HEAD_TITLE_SHORT = 'EF Cryptography Research';
...
+export const LEGENDRE_PRF_DATA_SOURCE = `${BOUNTIES_DATA_SOURCE}/legendre-prf`;
+export const ZK_HASH_DATA_SOURCE = `${BOUNTIES_DATA_SOURCE}/zk-hash.md`;
+export const RSA_DATA_SOURCE = `${BOUNTIES_DATA_SOURCE}/rsa`;
+export const MIMC_HASH_DATA_SOURCE = `${BOUNTIES_DATA_SOURCE}/mimc-hash-challenge.md`;
+
+// bounties URLs
+export const MIMC_HASH_BOUNTIES_URL = '/bounties/mimc-hash-challenge';
```
[the remark this answers is not in the document]
Comment (nhsz) [reply src/constants.ts:34]: I'm not sure how to do that and what would be the advantage

File: src/pages/bounties/legendre-prf/concrete-instance-bounties.tsx (hunk 2 above)
Comment (nhsz) [reply src/pages/bounties/legendre-prf/concrete-instance-bounties.tsx:1]: The file structure is different, some bounties have subpages, some don't, the logic would be way more complex.

File: src/pages/bounties/legendre-prf/algorithmic-bounties.tsx (hunk 1 above)
Comment (nhsz) [reply src/pages/bounties/legendre-prf/algorithmic-bounties.tsx:38]: fixing this

**Review by samajammin — APPROVED** [review]
👍

**Pull request merged**